packages/design-toolkit/src/hooks/coordinate-field/use-coordinate-copy.ts

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright 2025 Hypergiant Galactic Systems Inc. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+import { getLogger } from '@accelint/logger';
+import { useState } from 'react';
+
+const logger = getLogger({
+  enabled: process.env.NODE_ENV !== 'production',
+  level: 'debug',
+  prefix: '[CoordinateField]',
+  pretty: true,
+});
+
+import { getAllCoordinateFormats } from '../../components/coordinate-field/coordinate-utils';
+import type {
+  CoordinateSystem,
+  CoordinateValue,
+} from '../../components/coordinate-field/types';
+import type { UseTimeoutCleanupResult } from './use-timeout-cleanup';
+
+const COPY_FEEDBACK_DURATION_MS = 2000;
+
+export interface UseCoordinateCopyOptions {
+  currentValue: CoordinateValue | null;
+  validationErrors: string[];
+  isDisabled: boolean;
+  registerTimeout: UseTimeoutCleanupResult['registerTimeout'];
+}
+
+export interface UseCoordinateCopyResult {
+  copiedFormat: CoordinateSystem | null;
+  handleCopyFormat: (formatToCopy: CoordinateSystem) => Promise<void>;
+  isFormatButtonEnabled: boolean;
+}
+
+export function useCoordinateCopy({
+  currentValue,
+  validationErrors,
+  isDisabled,
+  registerTimeout,
+}: UseCoordinateCopyOptions): UseCoordinateCopyResult {
+  const [copiedFormat, setCopiedFormat] = useState<CoordinateSystem | null>(
+    null,
+  );
+
+  const fallbackCopyToClipboard = (text: string) => {
+    const textArea = document.createElement('textarea');
+    textArea.value = text;
+    textArea.style.position = 'fixed';
+    textArea.style.left = '-999999px';
+    textArea.style.top = '-999999px';
+    document.body.appendChild(textArea);
+    textArea.focus();
+    textArea.select();
+    try {
+      document.execCommand('copy');
+      setCopiedFormat(null);
```

Comment:
checking my understanding: should this instead set it to the `formatToCopy`?


```suggestion
      setCopiedFormat(formatToCopy);
```